Lecattar/pipeline scripts

lee0c opened the pull request:
Implements #35 with Azure Dev Ops Pipelines & adds local dev workflow for e2e test.

jsturtevant commented on hack/README.md:
```diff
@@ -0,0 +1,31 @@
+
+## Prerequisites
+
+In order to run the following, you need:
```
"in order to run make e2etest"

lee0c replied:
Check new wording/layout in CONTRIBUTING.md, but should be better?

jsturtevant commented on hack/README.md:
```diff
@@ -0,0 +1,31 @@
+| `SERVICEBUS_NAMESPACE` | Service bus namespace | Yes | No |
+| `SERVICEBUS_QUEUE_NAME` | Name of the service bus queue | Yes | Yes, defaults to `externalq` if not set |
+| `GOPATH` | Golang project directory | Yes | Yes, defaults to `$HOME/go` if not set |
+
```
Add example command to run here that outlines steps like building test image, pushing to repo and running make test-e2e

lee0c replied:
Haven't explicitly added example commands, but I think the flow in CONTRIBUTING.md makes it clear. Let me know if you still want the example

jsturtevant commented on hack/run-e2e.sh:
```diff
@@ -0,0 +1,48 @@
+kubectl config current-context
+JSONPATH='{range .items[*]}{@.metadata.name}{"\t"}Ready={@.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}'
+if kubectl get nodes -o jsonpath="$JSONPATH" | grep "Ready=False"; then 
+    exit 1
```
Leave a message for the user to set up cluster config properly letting them know it is in ./kube/config or can use custom  EXPORT KUBECONFIG=path

jsturtevant commented on hack/run-e2e.sh:
```diff
@@ -0,0 +1,48 @@
+set +o errexit 
+
+echo; echo "Running deployment scripts"
+cd $GOPATH/src/github.com/Azure/azure-k8s-metrics-adapter/.azdevops/2_deploy
```
move /.azdevops/2_deploy scripts to the hack folder.

jsturtevant replied:
consider using use `pushd` and `pop` over `cd`

lee0c replied:
From some brief googling it looks like the main plus of `pushd`/`popd` over `cd` is that it allows for more efficient file system navigation. Since it's being done in a script and not by a user, I will leave it as `cd` as it shouldn't matter in this case.

lee0c commented:
>     * Add checksums for files that are downloaded.

Added for helm, minikube, crictl. Removed kubectl download as it is actually already installed on build agents.

>     * Use hack/run-e2e.sh in build pipline in place of individual scripts.

Done, part of switching to this involved nuking the debug output in the build pipeline as it was happening after `hack/run-e2e.sh` was cleaning up the adapter deployment. Otherwise no major changes from this.

lee0c commented:
Add'l note on possible point of error in the future: right now, the deploy pipeline will fail if queued on its own rather than triggered by the image pipeline due to the way it gets the image tag variable. I'm looking into alternatives to this but the easy answer is just don't ever run deploy-pipeline on its own
